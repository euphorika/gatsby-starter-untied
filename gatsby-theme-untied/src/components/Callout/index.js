/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'

const Callout = ({ message, button }) => (
  <section
    sx={{
      textAlign: 'center',
      p: 4,
      mx: 4,
      my: 5,
      bg: 'accent',
      borderRadius: 1,
    }}
  >
    <h2
      sx={{
        fontSize: 3,
        color: '#f1f1f1',
        lineHeight: '1.2em',
        mb: 1,
        p: 0,
      }}
    >
      {' '}
      {message}{' '}
    </h2>
    <button
      sx={{
        bg: 'accent',
        fontSize: 3,
        border: 'none',
      }}
    >
      <Link
        sx={{
          variant: 'anchors.callout',
          fontWeight: 'bold',
          textDecoration: 'underline',
          color: '#f1f1f1',
        }}
        to={button.link}
      >
        {button.text}
      </Link>
    </button>
  </section>
)

Callout.propTypes = {
  message: PropTypes.string.isRequired,
  button: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
}
export default Callout
