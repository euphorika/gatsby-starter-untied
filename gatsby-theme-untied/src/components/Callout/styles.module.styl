@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'



.callout-section
  text-align: center
  padding-top: 25px
  padding-right: 25px
  padding-bottom: 25px
  padding-left: 25px
  margin: 50px 30px
  background-color: alert-color
  max-width: 100%
  border-radius: 8px


.message
  font-size: 20px
  color: secondary-color
  line-height: 1.2em
  margin-bottom: 5px
  padding: 0

.button
  background-color: alert-color
  font-size: 20px
  border: none

  .button-link
    color: secondary-color
    font-weight: 700

.text
    color: secondary-color
