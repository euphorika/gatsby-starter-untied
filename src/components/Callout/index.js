import React from 'react'

import styles from './styles.module.styl'

const Callout = () => (
  <section className={styles.calloutSection}>Callout</section>
)

export default Callout
