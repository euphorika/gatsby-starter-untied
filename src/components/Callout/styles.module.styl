@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.callout-section
  margin 0
