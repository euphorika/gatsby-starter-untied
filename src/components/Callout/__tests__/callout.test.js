import React from 'React'
import renderer from 'react-test-renderer'
import Callout from '../'

describe('Callout', () => {
  it('renders correctly', () => {
    const button = {
      text: 'Text',
      link: '/',
    }

    const callout = (
      <Callout message="Message" button={button}></Callout>
    )
    const tree = renderer.create(callout).toJSON()    
    expect(tree).toMatchSnapshot()
  })
})
