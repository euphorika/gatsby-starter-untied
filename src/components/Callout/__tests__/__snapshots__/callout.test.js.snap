// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout renders correctly 1`] = `
<section
  className="calloutSection"
>
  <button
    className="button"
  >
    <Link
      to="/"
    >
      <span
        className="text"
      >
         
        Text
      </span>
    </Link>
  </button>
  <h2
    className="message"
  >
     
    Message
     
  </h2>
</section>
`;
